source/Ship.cpp

```diff
@@ -3164,7 +3179,7 @@ bool Ship::CanCarry(const Ship &ship) const
 	for(const auto &it : escorts)
 	{
 		auto escort = it.lock();
-		if(escort && escort->attributes.Category() == category)
+		if(escort && escort->attributes.Category() == category && !IsShipInBays(escort))
```

Comment:
I feel that changing `BaysFree(...)` on line 3160 (3175 in the new situation) to `BaysTotal(...)` is a more efficient way of  fixing the bug, since the IsShipInBays-function-call introduces an additional loop per escort-ship, while changing line 3160 would actually reduce the amount of code that is run a tiny bit. (And this code runs in the real-time part of the game, so efficiency and performance are relevant here.)

Reply:
And maybe also add a check here if `(escort ==ship)`, to make sure that CanCarry(...) doesn't count the current ship double.

Reply:
Yes that would be cleaner and more performant.
I was unsure if existing escorts had priority over others, however it looks like the escorts just take the first available space.

Reply:
Thanks for the feedback @petervdmeer, I have pushed an update: https://github.com/endless-sky/endless-sky/pull/6091/commits/20979e5a75c459c669896e7a0a42c54fcff9a420